app/scripts/modules/core/src/managed/ToggleManagedResourceForApplication.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { memo, useState } from 'react';
+import ReactGA from 'react-ga';
+
+import { Application } from '../application';
+import { Button } from './Button';
+import {
+  Illustration,
+  IllustrationName,
+  IModalComponentProps,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  showModal,
+  ValidationMessage,
+} from '../presentation';
+import { ManagedWriter } from './ManagedWriter';
+
+const logClick = (label: string, application: string) =>
+  ReactGA.event({
+    category: 'Environments - toggle application management modal',
+    action: `${label} clicked`,
+    label: application,
+  });
+
+export interface IToggleManagedResourceForApplicationModalProps extends IModalComponentProps {
+  application: Application;
+}
+
+interface Error {
+  data: { error: string; message: string };
+}
+
+export const MDRunningStateDescription = () => (
+  <>
+    <p>
+      <span className="bold">
+        Careful! You’re about to stop Spinnaker from managing all resources in your application.
+      </span>
+      This feature should only be used if management is not working properly and manual intervention is required.{' '}
+      <a href="https://www.spinnaker.io/guides/user/managed-delivery" target="_blank">
+        Check our documentation for more information
+      </a>
+      .
+    </p>
+    <p>
+      Need to rollback?{' '}
+      <a href="https://www.spinnaker.io/guides/user/managed-delivery/pinning/" target="_blank">
+        Try pinning a version instead
+      </a>
+      .
+    </p>
+  </>
+);
+
+export const MDPausedStateDescription = () => {
+  return (
+    <p>
+      You’re about to resume management for this application. The latest good version approved for deployment will be
+      deployed to each environment, and any configuration changes made while disabled will take effect.
+    </p>
+  );
+};
+
+const ModalViewInfo = {
```

Comment:
if you decide to take action on either of the comments I left on `EnvironmentHeaderInfo`, probably worth applying those changes here too.